tools/vma_xls_extract.py

```
@@ -1,483 +1,281 @@
-                    break
-                cell_val = COLUMN_DTYPE_MAP[col_name](tools.util.empty_to_nan(cell_val))  # conversions
-                new_row[col_name] = cell_val
-                idx += 1
-            if done or all(pd.isna(v) for k, v in new_row.items() if k not in ['Common Units', 'Weight']):
-                last_row = r
-                break  # assume an empty row (except for Common Units) indicates no more sources
-            else:
-                df = df.append(new_row, ignore_index=True)
-        else:
-            raise Exception(f"No blank row detected in table. Either there are {max_sources} " +
-                    f"VMAs in table, the table is overfull, or there is some error in the code. "
-                    f"Cell: {source_id_cell}")
-
-        if (df['Weight'] == 0).all():
-            # Sometimes all weights are set to 0 instead of blank. Change them to be NaN.
-            df['Weight'] = df['Weight'].replace(0, np.nan)
-
-        # find the "Use Weight parameter past the last row of the VMA"
-        # Denise 7/21:  This is a little bit fragile.  If one table was missing Use Weight,
-        # it could pick up the Use Weight of the next one.  Probably not a likely scenario.
-        for r in range(last_row, last_row + 100):
-            if sheet.cell(r, tools.util.co("R")).value == 'Use weight?':
-                use_weight = tools.util.convert_bool(sheet.cell(r+1, tools.util.co("R")).value)
-                break
-            if sheet.cell(r, tools.util.co("S")).value == 'Use weight?':
-                use_weight = tools.util.convert_bool(sheet.cell(r+1, tools.util.co("S")).value)
-                break
+def get_vma_sheet(wb):
+    """Return the preferred VMA worksheet in this workbook"""
+    if "Variable Meta-analysis-DD" in wb.sheetnames:
+        return "Variable Meta-analysis-DD"
+    if "Variable Meta-analysis" in wb.sheetnames:
+        return "Variable Meta-analysis"
+    return None
```

callers don't seem ready for None. Maybe throw an exception (with the two desired names + the actual wb.sheetnames in the err str, so callers can see what kind of mismatch happened)

---

I will make it an error, I think that is a good call.
(I don't like the idea of listing available sheetnames, though, as there may be dozens, and that kind of verbosity obscures more than it illuminates IMHO.)